sunpy/visualization/animator/base.py

```diff
@@ -532,30 +536,36 @@ def _sanitize_axis_ranges(self, axis_ranges, data_shape):
                     axis_ranges[i] = edges_to_centers_nd(axis_ranges[i], i)
                 else:
                     raise ValueError(incompatible_axis_ranges_error_message(i))
-        # For each axis validate and translate the axis_ranges.
+
+        # For each slider axis validate and translate the axis_ranges.
+        def get_pixel_to_world_callable(array):
+            def pixel_to_world(pixel):
+                return array[pixel]
+            return pixel_to_world
```

Comment:
What is this code block doing...?

Reply:
This function now converts all the slider axes to functions, which are then called by the base animator when the slider is updated. So this is just moving the array slice into a function.

Reply:
Is there a reason it's a function defined in a function defined in a function?

Reply:
yes, because it uses `array` from the defining scope (which is called a closure)

Reply:
Still don't understand this but as I see you've thought it through a second time I trust it's fine ;)